### PR TITLE
feat(GMAP): overlay Positioning

### PR DESCRIPTION
**Pull request by dawsonpa**
affects: @rentpath/react-ui-core

Add Prop to the GMAP OverlayView called overlayPosition. It lets you set the absolute position of
the overlay on the map. If it is not provided then the default positioning is used.

### CODE REVIEW

File: packages/react-ui-core/src/Gmap/OverlayView.js [hunk 1]
```diff
@@ -111,18 +116,19 @@ export default class OverlayView extends PureComponent {
     let display = 'none'
 
     if (position) {
-      const projection = this.overlay.getProjection()
-
-      // Projection is undefined if map has not yet initialized
-      if (!projection) return
```
Comment (mpelzsherman) [inline packages/react-ui-core/src/Gmap/OverlayView.js]: Why was this removed?

File: packages/react-ui-core/src/Gmap/OverlayView.js [hunk 2]
```diff
@@ -111,18 +116,19 @@ export default class OverlayView extends PureComponent {
...
-
-      const { x, y } = projection.fromLatLngToDivPixel(position)
+      const { x, y } = this.overlay.getProjection().fromLatLngToDivPixel(position)
```
Comment (mpelzsherman) [inline packages/react-ui-core/src/Gmap/OverlayView.js]: Without the above check, this like could throw if `this.overlay.getProjection()` is undefined or null.

File: packages/react-ui-core/src/Gmap/OverlayView.js (hunk 1 above)
Comment (dawsonpa) [reply packages/react-ui-core/src/Gmap/OverlayView.js]: Yea good catch. It should not have been removed.

File: packages/react-ui-core/src/Gmap/OverlayView.js (hunk 2 above)
Comment (dawsonpa) [reply packages/react-ui-core/src/Gmap/OverlayView.js]: True

File: packages/react-ui-core/src/Gmap/OverlayView.js [hunk 3]
```diff
@@ -116,13 +121,19 @@ export default class OverlayView extends PureComponent {
       // Projection is undefined if map has not yet initialized
       if (!projection) return
 
-      const { x, y } = projection.fromLatLngToDivPixel(position)
+      const { x, y } = this.overlay.getProjection().fromLatLngToDivPixel(position)
```
Comment (mpelzsherman) [inline packages/react-ui-core/src/Gmap/OverlayView.js]: why not just use `projection` here? no need to call `this.overlay.getProjection()` again... is there?

**Comment by tadjohnston** [comment]
Closing, not needed anymore.
